src/MySqlConnector/Core/CachedProcedure.cs

```diff
@@ -205,6 +205,7 @@ internal static string ParseDataType(string sql, out bool unsigned, out int leng
 			sql = Regex.Replace(sql, " (CHARSET|CHARACTER SET) [A-Za-z0-9_]+", "", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
 			sql = Regex.Replace(sql, " (COLLATE) [A-Za-z0-9_]+", "", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
 			sql = Regex.Replace(sql, @"ENUM\s*\([^)]+\)", "ENUM", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
+			sql = Regex.Replace(sql, @"\s*NULL\s*$", "", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
```

Comment:
```suggestion
			sql = Regex.Replace(sql, @"\bNULL\s*$", "", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
```

The leading `\s*` feels both pointless (it can always be ignored) and dangerous (should there be _some_ kind of separator from any previous token); would `\b` be better?

Reply:
Alternatively, `\s+NULL\s*$` to require at least one space (which I assume SingleStore adds) and trim it off at the same time?

Reply:
Good idea - using `\s+` is the best option here.